### ISSUE TITLE
Torrents have to be stopped individually because of dht node cache

### ISSUE THREAD

**Issue by razaqq** [alanmcgovern/monotorrent#499]
When stopping a set of torrents with

```c#
// okay
foreach (TorrentManager tm in Engine.Torrents)
{
    await tm.StopAsync();
}

// doesnt work
await Task.WhenAll(Engine.Torrents.Select(tm => tm.StopAsync()));
```

i constantly run into the issue of
`The process cannot access the file 'C:\...\dht_nodes.cache' because it is being used by another process.`.

The error stems from this line
```c#
        async ReusableTasks.ReusableTask MaybeSaveDhtNodes ()
        {
            if (!Settings.AutoSaveLoadDhtCache)
                return;

            var nodes = await DhtEngine.SaveNodesAsync ();
            if (nodes.Length == 0)
                return;

            await Task.Run (() => {
                var savePath = Settings.GetDhtNodeCacheFilePath ();
                var parentDir = Path.GetDirectoryName (savePath);
                Directory.CreateDirectory (parentDir);
                File.WriteAllBytes (savePath, nodes);  // <-- file in use
            });
        }
```

This is with version `v2.0.1`

**Comment by alanmcgovern** [comment]
This is an interesting issue! This is should be relatively easy to fix. In the meantime you should be safe using the 'working' approach as a temporary workaround. 

If you're interested in contributing a fix, I believe the best option would be to only store the Dht node cache when the final torrent has stopped, and/or as part of ClientEngine.SaveStateAsync.

**Comment by alanmcgovern** [comment]
Fixed in master and 2.0! Keep an eye out for an update NuGet release!

**Comment by razaqq** [comment]
Thank you very much, it seems to be fixed now
